### PR TITLE
Added Fibonacci_Heap for DSA

### LINKED ISSUE #9095: Adding Fibonacci Heap data structure implementation codes
### Feature description

I think we should add implementation codes on fibonacci heaps in Python and various other languages to this repository. I didn't find any implementation of this data structure here

### PR DESCRIPTION
**Pull request by imSanko**
### Describe your change:

Added the Fibonacci heap in the DSA section 

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #9095".

### CODE REVIEW

File: data_structures/heap/fibonacci_heap.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+class FibonacciNode:
+    def __init__(self, key):
+        self.key = key
+        self.children = []
+        self.marked = False
+        self.degree = 0
+        self.parent = None
+
+class FibonacciHeap:
+    def __init__(self):
```
[the remark this answers is not in the document]
Comment (imSanko) [reply data_structures/heap/fibonacci_heap.py:11]: class FibonacciNode:
    def __init__(self, key) -> None:
        self.key = key
        self.children = []
        self.marked = False
        self.degree = 0
        self.parent = None

class FibonacciHeap:
    def __init__(self) -> None:
        self.trees = []
        self.least = None
        self.count = 0

File: data_structures/heap/fibonacci_heap.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            self.cascading_cut(parent)
+
+    def cut(self, node, parent):
+        parent.children.remove(node)
+        parent.degree -= 1
+        self.trees.append(node)
+        node.parent = None
+        node.marked = False
+
+    def cascading_cut(self, node):
```
[the remark this answers is not in the document]
Comment (imSanko) [reply data_structures/heap/fibonacci_heap.py:89]: class FibonacciNode:
    def __init__(self, key) -> None:
        self.key = key
        self.children = []
        self.marked = False
        self.degree = 0
        self.parent = None

class FibonacciHeap:
    def __init__(self) -> None:
        self.trees = []
        self.least = None
        self.count = 0

    # ... (other methods)

    def cascading_cut(self, node: FibonacciNode) -> None:
        """
        Perform cascading cuts to handle marked nodes and maintain heap properties.

        :param node: The node on which cascading cuts are applied.
        """
        parent = node.parent
        if parent is not None:
            if not node.marked:
                node.marked = True
            else:
                self.cut(node, parent)
                self.cascading_cut(parent)

**Review by imSanko — COMMENTED** [review]
Added reviews

**Comment by tianyizheng02** [comment]
Do not contribute multiple algorithms in one PR. Why did you check the checkbox if you didn't follow the instructions?
> - [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.

**Pull request closed without merging**